tests/integration/test_integration.py

```diff
@@ -1040,6 +1061,109 @@ def test_uniques_subset_within_complex_with_outputcheck_extralong(
     ), test_result.failure_message
 
 
+@pytest.mark.parametrize(
+    "data",
+    [
+        (
+            negation,
+            ["col_int", "col_varchar"],
+            [(0, "hi0"), (1, "hi0")],
+            0,
+            filternull_element_or_tuple_any,
+            False,
+            [output_processor_sort, functools.partial(output_processor_limit, limit=5)],
+            None,
+            None,
+            "column(s) 'col_int', 'col_varchar' has a fraction of 0.9997569866342649 > 0 DISTINCT values (8228 / 8230) not being an element of '[(0, 'hi0'), (1, 'hi0')]'. It has excess elements '[(2, 'hi1'), (3, 'hi2'), (5, 'hi3'), (6, 'hi4'), (8, 'hi5'), '<SHORTENED OUTPUT, displaying the first 5 / 8228 elements above>']' with counts [2, 2, 2, 2, 2, '<SHORTENED OUTPUT, displaying the first 5 / 8228 counts above>'].",
+        ),
+    ],
+)
+@pytest.mark.limit_memory(limit="60 MB")
+def test_memory_no_caching(engine, unique_table_largesize, data):
+    (
+        operation,
+        columns,
+        uniques,
+        max_relative_violations,
+        filter_func,
+        compare_distinct,
+        output_processors,
+        function,
+        condition,
+        failure_message_suffix,
+    ) = data
+
+    req = requirements.WithinRequirement.from_table(*unique_table_largesize)
+    for i in range(20):
+        req.add_uniques_subset_constraint(
```

Comment:
Why do we need to add 20 constraints instead of calling `test()` 20 times?

Reply:
If we just called test 20x, it would re-use the cached data—no out-of-memory would be triggered. This test is meant to mimic a typical use case with many memory-intensive constraints being checked sequentially.

Reply:
Ah I see, please add a comment pointing out `cache_size=0` and the purpose of this test

Reply:
Can't you also use the same query collector to check that nothing has been cached?

Reply:
Done.

Reply:
yes, I can additionally add it as well :) - Done